internal/services/server/server_test.go

```diff
@@ -216,12 +219,22 @@ func TestServerHandlers(t *testing.T) {
 		{"methodNotAllowedHandler", r, http.MethodDelete, "/", http.StatusMethodNotAllowed, "", nil, nil},
 		{"writeGetMetricHandler", r, http.MethodHead, "/", http.StatusMethodNotAllowed, "", nil, nil},
 	}
+	fmem, err := os.Create(`./../../../profiles/result.pprof`)
```

Comment:
This is not a good approach. Please review K6 https://github.com/grafana/k6?tab=readme-ov-file